recipes/openassetio/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+    }
+    short_paths = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "9",
+            "clang": "12",
+            "apple-clang": "12",
+        }
+
+    def configure(self):
+        if self.options.with_python:
+            if is_msvc(self):
+                # Required to create import .lib for building extension module.
+                self.options["cpython"].shared = True
+        else:
+            self.options.rm_safe("python_version")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("tomlplusplus/3.2.0")
+        if self.options.with_python:
+            # Avoid conflict with cmake vs cpython openssl dependency.
+            self.requires("openssl/1.1.1s")
+            # TODO: cpython requires ncurses/6.2 but no pre-built package exists.
+            self.requires(f"ncurses/6.3")
+            self.requires(f"cpython/{self.options.python_version}")
+            self.requires("pybind11/2.10.1")
+
+    def validate(self):
+        if is_apple_os(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} does not support MacOS at this time"
+            )
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+
+        if is_msvc(self) and not self.dependencies["cpython"].options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} requires cpython:shared=True when using MSVC compiler")
+
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.25.1")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        tc.variables["OPENASSETIO_ENABLE_TESTS"] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+        tc.variables["OPENASSETIO_GLIBCXX_USE_CXX11_ABI"] = self.settings.get_safe("compiler.libcxx") == "libstdc++11"
+
+        tc.variables["OPENASSETIO_ENABLE_PYTHON"] = self.options.with_python
+        if self.options.with_python:
+            tc.variables["Python_EXECUTABLE"] = self._python_exe
+            if is_msvc(self):
+                tc.variables["Python_LIBRARY"] = self._python_windows_lib
+
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
```

Comment:
Buidl should be build scope ?

```suggestion
        tc.generate()
```